store/migration/manager.go

```diff
@@ -134,14 +134,16 @@ func (m *Manager) writeChangeset() error {
 		}
 
 		batch := m.db.NewBatch()
-		defer batch.Close()
 
 		if err := batch.Set(csKey, csBytes); err != nil {
 			return fmt.Errorf("failed to write changeset to db.Batch: %w", err)
 		}
 		if err := batch.Write(); err != nil {
 			return fmt.Errorf("failed to write changeset to db: %w", err)
 		}
+		if err := batch.Close(); err != nil {
+			return fmt.Errorf("failed to close batch: %w", err)
+		}
```

Comment:
The problem with the current change is that if batch.Set or batch.Write fail, then batch.Close() will now not be invoked. I do understand the importance of removing defers in the loop to avoid lingering resources that won't be cleaned up until a panic or a return, but the better fix for this would be making an anonymous closure so
```go
// Invoking this code in a closure so that defer is called immediately on return
// yet not in the for-loop which can leave resource lingering.
err := func() error {
     defer batch.Close()

     if err := batch.Set(csKey, csBytes); err != nil {
	return fmt.Errorf("failed to write changeset to db.Batch: %w", err)
     }
     if err := batch.Write(); err != nil {
	return fmt.Errorf("failed to write changeset to db: %w", err)
     }
}()

if err != nil {
    return err
}
```